_posts/2023-08-01-tmt-cross-project-testing.md

```diff
@@ -0,0 +1,131 @@
+
+![packit allowed forge projects](/images/packit-allowed-forge-projects.png)
+
+Take a look at [Cockpit's "main-builds" COPR](https://copr.fedorainfracloud.org/coprs/g/cockpit/main-builds/) for an example.
+
+Then configure `packit.yaml` in your project to do a [build for each commit](https://packit.dev/docs/configuration/upstream/copr_build#supported-triggers) that lands on your "main" branch; that may have a different name for your project of course. For example, [cockpit-podman did this](https://github.com/cockpit-project/cockpit-podman/pull/1365):
+
+```yaml
+- job: copr_build
+  trigger: commit
+  branch: "^main$"
+  owner: "@cockpit"
+  project: "main-builds"
+  preserve_project: True
+```
+
+After landing that, check your COPR's "Builds" page that you actually get a successful package build. The first run will fail, you will get a notification email and have to approve the "packit" COPR user to trigger builds in your COPR.
+
+## Enable your test in a dependency project
+
+This is the step which requires buy-in from the project to which you want to add your tests. That conversation needs to include some agreement how test failures are handled, who will look at them and in which time frame, and similar commitments. Possibly also a link to this blog post to provide the background and motivation 😀.
+
+Add a new `plans/yourproject.fmf` test plan to the dependency which selects the
+tests you want to run from your project. Unfortunately you cannot yet [auto-import all plans](https://github.com/teemtee/tmt/issues/1770), but first of all the plan structure doesn't tend to change often, and second this may actually be useful to select a subset of tests that apply to the tested dependency.
+
+So this is mostly a copy of your project's main test plan, with one modification: they should not run by default, but only in a "revdeps" tmt context. For triggering cockpit-podman's tests, this can look like this:
+```yaml
+# reverse dependency test
+enabled: false
+
+adjust+:
+  when: revdeps == yes
+  enabled: true
+
+discover:
+    how: fmf
+    url: https://github.com/cockpit-project/cockpit-podman
+    ref: "main"
+execute:
+    how: tmt
+
+/podman-system:
+    summary: Run cockpit-podman system tests
+    discover+:
+        test: /test/browser/system
+
+# ... possibly more test plans here
+```
+
+Conversely, you should mark the main test plan for the dependency project to *not* run in the "revdeps" scenario, to keep them apart in the PR statuses and log views:
+
+```yaml
+adjust+:
+  when: revdeps == yes
+  enabled: false
+```
+
+Finally, plug it all together: Tell `packit.yaml` to run the "revdeps" context test plan on current Fedora against your "builds from main" COPR, by specifying [tf_extra_params](https://packit.dev/docs/configuration/upstream/tests#optional-parameters). You can of course choose more/different `target`s, as long as your COPR has matching builds for them:
```

Comment:
does it make sense to mention the usage of `identifier` that is then reflected in the status name or is that too much details?

Reply:
Yes, it does, see above. I updated this paragraph.